platform/ios/src/MGLMapView.mm

```diff
@@ -3550,6 +3555,10 @@ - (MGLAnnotationTag)annotationTagAtPoint:(CGPoint)point persistingResults:(BOOL)
         auto end = std::remove_if(nearbyAnnotations.begin(), nearbyAnnotations.end(),
                                   [&](const MGLAnnotationTag annotationTag)
         {
+            if (annotationTag == MGLAnnotationTagNotFound) {
```

Comment:
Going to answer my own question here: `-annotationTagsInRect:` wouldn’t know about the user location annotation, so it should never return `MGLAnnotationTagNotFound`. I still think we should either rely on the assertion below or assert here about `MGLAnnotationTagNotFound`, in case some sort of internal consistency leads it to show up in `nearbyAnnotations`.